database/src/postgres/models.rs

```diff
@@ -0,0 +1,569 @@
+use crate::schema::*;
+use diesel::prelude::*;
+use diesel_async::RunQueryDsl;
+
+/// State-indexer tables
+#[derive(Insertable, Queryable, Selectable)]
+#[diesel(table_name = state_changes_data)]
+pub struct StateChangesData {
+    pub account_id: String,
+    pub block_height: bigdecimal::BigDecimal,
+    pub block_hash: String,
+    pub data_key: String,
+    pub data_value: Option<Vec<u8>>,
+}
+
+impl StateChangesData {
+    pub async fn save(&self, mut conn: crate::postgres::PgAsyncConn) -> anyhow::Result<()> {
```

Comment:
The verb "save" has two meanings in this context:
- Save as new in DB (insert)
- Save after changes (update)

Please, consider renaming to something straightforward for our own sake.